web/src/App.tsx

```diff
@@ -53,9 +53,8 @@ const DEFAULT_SETTINGS = {
 
 const UNREAL_BLOOM_PASS = new UnrealBloomPass()
 
-const { xGraph: X_GRAPH, nodes: NODES, fileTree: FILE_TREE } = buildXGraph(Data.__INLINE_DATA)
-
-function App () {
+function GraphExplorer({ graphData }: { graphData: Graph }) {
+  const { xGraph: X_GRAPH, nodes: NODES, fileTree: FILE_TREE } = useMemo(() => buildXGraph(graphData), [graphData])
```

Comment:
As these are no longer global constants, it would make sense to keep the casing consistent with other local constants:
```suggestion
  const { xGraph, nodes, fileTree } = useMemo(() => buildXGraph(graphData), [graphData])
```